examples/gno.land/p/demo/auction/auction.gno

```diff
@@ -0,0 +1,140 @@
+package auction
+
+import (
+	"std"
+	"strconv"
+	"time"
+
+	"gno.land/p/demo/avl"
+)
+
+// Main struct
+type Auction struct {
+	title       string
+	owner       std.Address
+	description string
+	begin       time.Time
+	end         time.Time
+	price       uint64
+	bids        *avl.Tree // key: std.Address, value: Bid
+}
+
+type Bid struct {
+	bidder std.Address
+	amount uint64
+}
+
+type EventEnd struct {
+	winner std.Address
+	amount uint64
+}
+
+func NewAuction(
+	title string,
+	owner std.Address,
+	description string,
+	begin time.Time,
```

Comment:
What do you think about having `begin` actually be `time.Now`?
This way you can decrease the param. It's only assuming you don't have auctions that start in the future, given that you emit an event when this constructor exits

`end` can be `time.Duration` if this is your preference.

Reply:
I've updated the package with three auction states: “upcoming”, “ongoing”, and “closed”. That was my idea, so I could have auctions starting in the future.